craftassist/agent/dialogue_objects/mc_interpreter.py

```diff
@@ -380,32 +379,34 @@ def new_tasks():
                 repeat_dir=repeat_dir,
                 padding=padding,
             )
-            print("origin from dig", origin, "offsets", offsets)
-
             # add dig tasks in a loop
             tasks_todo = []
             for offset in offsets:
                 og = np.array(origin) + offset
                 t = self.task_objects["dig"](self.agent, {"origin": og, "action_dict": d, **attrs})
-                print("append task:", t, og, d, attrs)
                 tasks_todo.append(t)
-            return list(reversed(tasks_todo))
+            return maybe_task_list_to_control_block(tasks_todo, self.agent)
 
-        print("inside dig, dict", d)
         if "stop_condition" in d:
-            print("stop condition", d["stop_condition"])
             condition = self.subinterpret["condition"](self, speaker, d["stop_condition"])
-            self.append_new_task(
-                self.task_objects["loop"],
-                {"new_tasks_fn": new_tasks, "stop_condition": condition, "action_dict": d},
+            return (
+                [
+                    self.task_objects["control"](
```

Comment:
this will need to be updated with the spec change since we don't have the "control" key anymore ?

Reply:
this is not part of spec:  its the Task Object that implements control (e.g. a ControlBlock).  see https://github.com/facebookresearch/droidlet/blob/812c4f63b7e0804a6aa44ed206de9445459240e9/craftassist/agent/dialogue_objects/mc_interpreter.py#L93